internal/pkg/deploy/upload/asset/asset_test.go

```diff
@@ -4,35 +4,47 @@
 package asset
 
 import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/hex"
 	"errors"
 	"fmt"
 	"io"
+	"path/filepath"
 	"testing"
 
 	"github.com/spf13/afero"
 	"github.com/stretchr/testify/require"
 )
 
 type fakeS3 struct {
-	objects map[string]string
-	err     error
+	err error
 }
 
-func (f *fakeS3) UploadFunc() func(string, io.Reader) (string, error) {
-	return func(key string, dat io.Reader) (url string, err error) {
-		if f.err != nil {
-			return "", f.err
-		}
-		url, ok := f.objects[key]
-		if !ok {
-			return "", fmt.Errorf("key %q does not exist in fakeS3", key)
+func (f *fakeS3) Upload(bucket, key string, data io.Reader) (string, error) {
+	return "", f.err
+}
+
+func Test_UploadToCache(t *testing.T) {
```

Comment:
shall we make more test cases when 1) source is a dir and dest is not set 2) source is a file and dest is not set etc.

Reply:
labelled the existing test cases and added a test for the missing case! (source=file, dest set)